### PR TITLE
Migrate `vtctlclient InitShardMaster` => `vtctldclient InitShardPrimary`

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This implements the `InitShardMaster` command in the VtctldServer, renamed as `InitShardPrimary` to take advantage of the easy opportunity to rename without backwards compatibility considerations.

Some other decisions made in this PR to call out:
- I did no refactor or variable renames on the function, because I didn't want to add complications to the initial migration.
- The old version had no tests, including endtoend tests, so I added a couple simple cases. Any further advice on testing is welcome.
- **Important** - I included the well-known Duration type to handle the `--wait-replicas-timeout` flag, which requires inclusion of the standard proto definitions that ship with `protoc`. These _usually_ (and by default I think) end up in `/usr/local/include`, so I've taken the (possibly naive!) simple step of updating `make proto` to add that include path. I'd like to talk about ideas about how to ensure this doesn't break existing developer's workflows, and whether this is portable enough. Like, we could, for example, call this good enough and add a note to the proto readme to ensure you have these files in the right path, or we could make the tooling more dynamic.

## Related Issue(s)

- #7058 
- #7219

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

## Deployment Notes

None

## Impacted Areas in Vitess
Components that this PR will affect:

- [ ]  Query Serving
- [ ]  VReplication
- [x]  Cluster Management
- [ ]  Build

### CODE REVIEW

File: proto/vtctlservice.proto [hunk 1]
```diff
@@ -35,4 +35,11 @@ service Vtctld {
   rpc GetKeyspace(vtctldata.GetKeyspaceRequest) returns (vtctldata.GetKeyspaceResponse) {};
   // GetKeyspaces returns the keyspace struct of all keyspaces in the topo.
   rpc GetKeyspaces(vtctldata.GetKeyspacesRequest) returns (vtctldata.GetKeyspacesResponse) {};
+  // InitShardPrimary sets the initial primary for a shard. Will make all other
+  // tablets in the shard replicas of the provided primary.
+  //
+  // WARNING: This could cause data loss on an already replicating shard.
+  // PlannedReparentShard or EmergencyReparentShard should be used in those
+  // cases instead.
+  rpc InitShardPrimary(vtctldata.InitShardPrimaryRequest) returns (vtctldata.InitShardPrimaryResponse) {};
```
Comment (ajm188) [inline proto/vtctlservice.proto:44]: I'm genuinely unsure what the right API here is. Is all we want the stream of updates? Maybe this is a case where we actually want just a `stream logutil.Event` response 🤔

**Review by doeg — APPROVED** [review]
The golang parts look good to me! Thank you for doing the rename (and adding tests too, so nice!)

You'll fur sure want Planetscale signoff on the rest, of course. 🎉

**Comment by deepthi** [comment]
> I included the well-known Duration type to handle the --wait-replicas-timeout flag, which requires inclusion of the standard proto definitions that ship with protoc. These usually (and by default I think) end up in /usr/local/include, so I've taken the (possibly naive!) simple step of updating make proto to add that include path. I'd like to talk about ideas about how to ensure this doesn't break existing developer's workflows, and whether this is portable enough. Like, we could, for example, call this good enough and add a note to the proto readme to ensure you have these files in the right path, or we could make the tooling more dynamic.

In my local environment, the standard protos are under `vitess.io/vitess/dist/vt-protoc-3.6.1/include`. That is where bootstrap installs them.

**Comment by ajm188** [comment]
Working updated `make proto`:

```
❯ make proto
Tue Jan 5 19:44:17 EST 2021: Installing minimal dependencies
skipping protoc install. remove /Users/amason/work/vitess/dist/vt-protoc-3.6.1 to force re-install.
/usr/local/bin/etcd
etcd already installed
skipping k3s install. remove /Users/amason/work/vitess/dist/k3s to force re-install.

bootstrap finished - run 'make build' to compile
Tue Jan 5 19:44:57 EST 2021: Compiling proto definitions
```

File: proto/vtctlservice.proto (hunk 1 above)
Comment (deepthi) [reply proto/vtctlservice.proto:46]: I don't know either :) What you have is fine for now and we can fix forward if we decide otherwise.

**Review by deepthi — APPROVED** [review]
LGTM

File: go/vt/proto/automationservice/automationservice.pb.go [hunk 2]
```diff
@@ -12,7 +12,6 @@ import (
 	grpc "google.golang.org/grpc"
 	codes "google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
-
```
Comment (ajm188) [inline go/vt/proto/automationservice/automationservice.pb.go:15]: it looks like the new `goimports` style conflicts with how `protoc-gen-go` generates files

Comment (ajm188) [reply go/vt/proto/automationservice/automationservice.pb.go:15]: Fixes for this in 602dd69 (git hook) and 67ae893 (linter config)

**Comment by rohit-nayak-ps** [comment]
@ajm188, I reran a flaky test that had failed and now all tests are passing. Is this ready to be merged?

**Comment by ajm188** [comment]
Yep should be good to go!

**Pull request merged**